models/datasets/reservoir/ethereum/tokens_beta.sql

```diff
@@ -0,0 +1,41 @@
+{{ config(
+    tags = ['dunesql'],
+    schema = 'reservoir',
+    alias = alias('tokens_beta'),
+    post_hook = '{{ expose_dataset(\'["ethereum"]\',
+                \'[""]\') }}'
+    )
+}}
+select
+    t.created_at,
+    t.collection_id,
+    _unsafe_0x_or_base58_to_varbinary(t.contract) as contract,
+    _unsafe_0x_or_base58_to_varbinary(t.floor_ask_maker) as floor_ask_maker,
+    _unsafe_0x_or_base58_to_varbinary(t.floor_ask_id) as floor_ask_id,
+    _unsafe_0x_or_base58_to_varbinary(t.owner) as owner,
+    t.description,
+    t.floor_ask_source,
+    t.floor_ask_valid_from,
+    t.floor_ask_valid_to,
+    t.floor_ask_value,
+    t.floor_ask_value / 1e18 as floor_ask_value_decimal,
+    t.id,
+    t.last_sale_timestamp,
+    t.last_sale_value,
+    t.last_sale_value / 1e18 as last_sale_value_decimal,
+    t.name,
+    t.token_id,
+    t.updated_at
+from reservoir.tokens_0010 t
+    inner join (
+        select
+            id,
+            max(updated_at) as recent_updated_at,
+            max(filename) as last_filename
+        from
+            reservoir.tokens_0010
```

Comment:
we should avoid hardcoded source objects -- is the plan to establish these as sources? then introduce a source yaml file to maintain, use source jinja syntax here?

Reply:
we lose the capability for dbt lineage without `source()` or `ref()` usage

this is experimental, so if you want to proceed and potentially revisit later, feel free. it's just not something that is done currently across spellbook